vm.c

```diff
@@ -25,14 +26,22 @@
 
 #define END_OPCODES
 
-#define VM_GOTO(n) \
-    pc = n;        \
+#define VM_GOTO(n)  \
+    if (OPCODE.op1.type == LABEL && n == -1) {  \
+        printf("Error: Undefined label \"%s\".\n", OPCODE.op1.label); \
```

Comment:
Since macro `FATAL` is introduced in header `private.h`, you can rewrite with it.